src/wallet/wallet.cpp

```diff
@@ -1155,6 +1155,30 @@ void CWallet::BlockConnected(const std::shared_ptr<const CBlock>& pblock, const
     for (size_t i = 0; i < pblock->vtx.size(); i++) {
         SyncTransaction(pblock->vtx[i], pindex, i);
     }
+
+
+    if (pindex) {
+        // Only notify UI if this transaction is in this wallet
+        if (hashPrevBestCoinbase.IsNull()) {
+            // For correctness we scan over the entire wallet, looking for
+            // the previous block's coinbase, just in case it is ours, so
+            // that we can notify the UI that it should now be displayed.
```

Comment:
I agree with @ryanofsky's suggested code change. That structure seems clearer to me.

I think the commit message can be shortened and the reference to the previous bug removed. Something like:

```
This does a full mapWallet loop after the first block is connected
after restart. This is so NotifyTransactionChanged() is called for
the coinbase transaction of the previous best block if it belongs to us.
```

seems clearer to me.